### PR TITLE
Project goals frontend implementation 

### PR DESCRIPTION
**Pull request by Agill-Sheron**
https://linear.app/geyser/issue/GYS-6668/project-goals-frontend-implementation

### CODE REVIEW

File: src/components/icons/svg/SatoshiIconCircled.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { Icon, IconProps } from '@chakra-ui/react'
+
+export interface SatoshiIconCircledProps extends IconProps {
+  size?: 'lg' | 'md' | 'sm'
+}
```
Comment (sajald77) [inline src/components/icons/svg/SatoshiIconCircled.tsx:5]: You're using this in a few places, maybe standardize this. 
You can check it these are the sizes in the design system with vlad, and if so. you can add these to the Icon theme for chakra. 

If not, a normal defined type should be okay too.

File: src/graphql/mutations/project.ts [hunk 2]
```diff
@@ -95,30 +95,32 @@ export const MUTATION_DELETE_PROJECT_REWARD = gql`
 `
 
 export const MUTATION_CREATE_PROJECT_MILESTONE = gql`
-  mutation CreateProjectMilestone($input: CreateProjectMilestoneInput) {
-    createProjectMilestone(input: $input) {
+  mutation ProjectGoalCreate($input: ProjectGoalCreateInput!) {
```
Comment (sajald77) [inline src/graphql/mutations/project.ts:98]: might be a good idea to update the mutation name to: MUTATION_CREATE_PROJECT_GOAL

File: src/modules/project/context/ProjectProvider.tsx [hunk 3]
```diff
@@ -207,7 +206,7 @@ export const ProjectProvider = ({ projectId, children }: { children: React.React
       {project && isProjectOwner && (
         <>
           <ProjectCreatorModal {...creatorModal} />
-          <MilestoneAdditionModal {...milestonesModal} onSubmit={onMilestonesSubmit} project={project} />
+          {/* <MilestoneAdditionModal {...milestonesModal} onSubmit={onMilestonesSubmit} project={project} /> */}
```
Comment (sajald77) [inline src/modules/project/context/ProjectProvider.tsx:209]: do we have a new page for goal creation ? or a different modal?

File: src/modules/project/pages/projectView/views/projectActivityPanel/screens/info/hooks/useProjectDefaultGoal.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    const amount = getUSDAmount(project?.balance as Satoshis)
+    if (amount < 1) return 'less than $1'
+    return `$${commaFormatted(Math.round(amount))}`
+  }, [getUSDAmount, project?.balance])
+
+  const formattedSatsAmount = useCallback(() => {
+    const amount = getSatoshisFromUSDCents(priorityGoal?.amountContributed as USDCents)
+    if (amount < 1) return 'less than 1'
+    return `${commaFormatted(Math.round(amount))} sats`
+  }, [getSatoshisFromUSDCents, priorityGoal?.amountContributed])
```
Comment (sajald77) [inline src/modules/project/pages/projectView/views/projectActivityPanel/screens/info/hooks/useProjectDefaultGoal.tsx:54]: we could create utils functions for those. These are used in many places.

check useFundCalc, hook as well. There are some similarities. We could integrate this to that, possibly. 

If not, new functions would be okay too.

File: src/modules/project/pages/projectView/views/projectMainBody/components/GoalInProgress.tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+          {goal.description}
+        </Body1>
+      </HStack>
+      <HStack display="flex" alignItems="flex-start" width="100%"></HStack>
+      <Box
+        display="flex"
+        flexDirection={isMobile ? 'column' : 'row'}
+        alignItems="center"
+        width="100%"
+        gap={isMobile ? 2 : 10}
```
Comment (sajald77) [inline src/modules/project/pages/projectView/views/projectMainBody/components/GoalInProgress.tsx:114]:
```suggestion
        gap={{ base: 2, lg: 10}}
```

**Review by sajald77 — COMMENTED** [review]
Good work here Malik. Looks like it's coming together. I've added a few comments and suggestions. Will take a look again tomorrow. Looks like it's close.

File: src/modules/project/context/ProjectProvider.tsx (hunk 3 above)
Comment (Agill-Sheron) [reply src/modules/project/context/ProjectProvider.tsx:209]: We have a new modal for project creation GoalModal ! The Milestones were so embedded in in the project that it was not worth it to refactor it to goals.

**Pull request merged**